tests/e2e/test/e2e/util.go

```
@@ -643,21 +701,24 @@ func validateRepo(serviceName string, prefixFilter string, checkSignature bool)
-	var skopeooutStr string
+	// Retain stdout AND stderr for every attempt and log the real error. skopeo
+	// reports authentication/network failures on stderr, so capturing only stdout
+	// (as before) hid the actual reason for a failed copy and made a registry
+	// problem indistinguishable from a test defect.
+	var lastErr string
 	waitErr := wait.Poll(30*time.Second, 180*time.Second, func() (bool, error) {
-		skopeoout, err := exec.Command("bash", "-c", skopeoCommand).Output()
-		if err != nil {
-			e2e.Logf("copy failed, retrying...")
-			skopeooutStr = string(skopeoout)
+		cmd := exec.Command("bash", "-c", skopeoCommand)
```

golangci-lint (2.13.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)